src/BSON/Timestamp.c

```diff
@@ -214,6 +215,43 @@ PHP_METHOD(Timestamp, __toString)
 }
 /* }}} */
 
+/* {{{ proto array Timestamp::jsonSerialize()
+*/
+PHP_METHOD(Timestamp, jsonSerialize)
+{
+	php_phongo_timestamp_t *intern;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	intern = Z_TIMESTAMP_OBJ_P(getThis());
+
+	array_init_size(return_value, 1);
+
+#if PHP_VERSION_ID >= 70000
+	{
+		zval ts;
+
+		array_init_size(&ts, 2);
+		ADD_ASSOC_LONG_EX(&ts, "t", intern->timestamp);
+		ADD_ASSOC_LONG_EX(&ts, "i", intern->increment);
```

Comment:
The Extended JSON spec currently states that a Timestamp should yield:

```
{"$timestamp": {"t": integer, "i": integer}}
```

We're still discussing whether these values should be single strings or perhaps `{"$numberLong": string}` objects. We cannot guarantee portability with integers, since the values may fall outside the range of 32-bit signed integers (both values are unsigned 32-bit integers).